src/textual/app.py

```diff
@@ -1355,8 +1362,8 @@ def call_from_thread(
 
         async def run_callback() -> CallThreadReturnType:
             """Run the callback, set the result or error on the future."""
-            self._set_active()
-            return await invoke(callback_with_args)
+            with self._context():
```

Comment:
An interesting trick, I might use that in some of my code. Thanks for the tip in the tweet!